src/flamegraph/merge.rs

```diff
@@ -148,8 +172,29 @@ where
             iter::once("").chain(last.split(';')),
             None,
             time,
+            delta,
         );
     }
 
-    (frames, time, ignored)
+    (frames, time, ignored, delta_max)
+}
+
+// Parse and remove the number of samples from the end of a line.
+fn parse_nsamples(line: &mut &str) -> Option<usize> {
+    if let Some(samplesi) = line.rfind(' ') {
+        let mut samples = &line[(samplesi + 1)..];
+        // strip fractional part (if any);
+        // foobar 1.klwdjlakdj
+        if let Some(doti) = samples.find('.') {
+            samples = &samples[..doti];
+        }
+        if let Ok(nsamples) = samples.parse::<usize>() {
```

Comment:
This can also become an `Option` using `samples.parse::<usize>().ok()`, at which point you can use `?` for this too!